src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -1331,7 +1362,7 @@ void BlockchainLMDB::open(const std::string& filename, const int db_flags)
 
   bool compatible = true;
 
-  MDB_val_copy<const char*> k("version");
+  MDB_val_nocopy<const char*> k("version");
```

Comment:
Don't need strlen(foo)+1 here. Just sizeof(foo) for string constants is all you need. Instead of a template you could get away with a plain CPP macro, similar to MDB_val_set(). There's no need for any runtime computation here, it can all be resolved at compile-time.

Reply:
Dangerous if you pass a var though, you end up with 8.